libraries/ESP8266WebServer/src/Parsing-impl.h

```diff
@@ -179,7 +181,7 @@ bool ESP8266WebServer::_parseRequest(WiFiClient& client) {
     String plainBuf;
     if (   !isForm
         && // read content into plainBuf
-           (   !readBytesWithTimeout(client, contentLength, plainBuf, HTTP_MAX_POST_WAIT)
+           (   !readBytesWithTimeout<ServerType>(client, contentLength, plainBuf, HTTP_MAX_POST_WAIT)
```

Comment:
This is a template function, there should be no need to explicitly specify the template arg due to automatic deduction (simplifies reading):
```cpp
           (   !readBytesWithTimeout(client, contentLength, plainBuf, HTTP_MAX_POST_WAIT)
```

Reply:
You'd think there'd be no reason, but the template arg really is needed there to make GCC happy:
````
In file included from /tmp/ci/arduino_ide/hardware/esp8266com/esp8266/libraries/ESP8266WebServer/src/ESP8266WebServer.h:213:0,
                 from /tmp/ci/libraries/DNSServer/examples/CaptivePortalAdvanced/CaptivePortalAdvanced.ino:3:
/tmp/ci/arduino_ide/hardware/esp8266com/esp8266/libraries/ESP8266WebServer/src/Parsing-impl.h: In instantiation of 'bool esp8266webserver::ESP8266WebServerTemplate<ServerType>::_parseRequest(esp8266webserver::ESP8266WebServerTemplate<ServerType>::ClientType&) [with ServerType = WiFiServer; esp8266webserver::ESP8266WebServerTemplate<ServerType>::ClientType = WiFiClient]':
/tmp/ci/arduino_ide/hardware/esp8266com/esp8266/libraries/ESP8266WebServer/src/ESP8266WebServer-impl.h:330:41:   required from 'void esp8266webserver::ESP8266WebServerTemplate<ServerType>::handleClient() [with ServerType = WiFiServer]'
/tmp/ci/libraries/DNSServer/examples/CaptivePortalAdvanced/CaptivePortalAdvanced.ino:141:23:   required from here
/tmp/ci/arduino_ide/hardware/esp8266com/esp8266/libraries/ESP8266WebServer/src/Parsing-impl.h:184:89: error: no matching function for call to 'readBytesWithTimeout(esp8266webserver::ESP8266WebServerTemplate<WiFiServer>::ClientType&, uint32_t&, String&, int)'
            (   !readBytesWithTimeout(client, contentLength, plainBuf, HTTP_MAX_POST_WAIT)
                                                                                         ^
/tmp/ci/arduino_ide/hardware/esp8266com/esp8266/libraries/ESP8266WebServer/src/Parsing-impl.h:184:89: note: candidate is:
/tmp/ci/arduino_ide/hardware/esp8266com/esp8266/libraries/ESP8266WebServer/src/Parsing-impl.h:43:13: note: template<class ServerType> bool esp8266webserver::readBytesWithTimeout(typename ServerType::ClientType&, size_t, String&, int)
 static bool readBytesWithTimeout(typename ServerType::ClientType& client, size_t maxLength, String& data, int timeout_ms)
             ^
/tmp/ci/arduino_ide/hardware/esp8266com/esp8266/libraries/ESP8266WebServer/src/Parsing-impl.h:43:13: note:   template argument deduction/substitution failed:
/tmp/ci/arduino_ide/hardware/esp8266com/esp8266/libraries/ESP8266WebServer/src/Parsing-impl.h:184:89: note:   couldn't deduce template parameter 'ServerType'
            (   !readBytesWithTimeout(client, contentLength, plainBuf, HTTP_MAX_POST_WAIT)
                                                                                         ^
````

Reply:
You're right, it's a dependent type, not worth it